upup/models/bindata.go

```diff
@@ -14233,7 +14233,7 @@ data:
           force_tcp
         }
         prometheus :9253
-        health {{ KubeDNS.NodeLocalDNS.LocalIP }}:8080
+        health {{ KubeDNS.NodeLocalDNS.LocalIP }}:{{ NodeLocalDNSHealthCheck }}
```

Comment:
I was wondering if this should be configurable in the cluster spec. Then you can use the well-known port as default in `components/kubedns.go` instead of adding the template function. Just a thought.

Reply:
What is the use case for making this port configurable?

Incidentally, I wonder if we could have kops automatically default the value of `clusterDNS` in `kubelet` and `masterKubelet` to `169.254.20.10` when needed rather than just mentioning it in documentation.

Reply:
Well. Both can conflict with other things, so it isn't unreasonable to have this configurable, I think.

Reply:
Our approach is to reserve ports in `wellknownports`. If anything else runs in host network and conflicts with those, the other things are what have to move.

Reply:
It is still a breaking change in kops.  That said, it is something we need to do, so maybe we should just reserve a range of ports for kops usage.

I think it's a good idea also to make it as easy as possible to turn on NodeLocalDNS as possible.  Can we change the defaulting logic when nodelocal dns is enabled, but continue to honor an explicit value if the user sets one?